build-tools/packages/version-tools/src/internalVersionScheme.ts

```diff
@@ -425,6 +455,11 @@ export function detectInternalVersionConstraintType(range: string): "minor" | "p
 		throw new Error(`Range ${range} is not a Fluid internal version range.`);
 	}
 
+	// These cases are not expected to be positive; nor are robust.
+	// internalPrerelease only applies to versions, which won't have ~ or ^.
+	// internal ranges are restricted to uses of `>=` even though semver would permit more.
+	// For prerelease spec (internal range), ~ and ^ are interpreted as >= by semver.
+	// Why not just use the bump and checks all the time?
 	if (range.startsWith("~")) {
```

Comment:
`~` and `^` are checked because there are/were cases where an internal version was naively prepended with one of those, even though it's not valid technically. What should this function do if passed a value like `^2.0.0-internal.1.0.0`?

Reply:
Are there cases now? Why aren't they getting fixed?
The comment I added is not really correct. But I don't think such cases should be treated as internal because the semver meaning of them is treat ~ and ^ as >=(given spec) to ^(given spec less prerelease), at least as far as I can ascertain.
I would say they should just be rejected - throw as invalid.

Reply:
After looking at the uses of this, I go back to my original comment as correct. This is not reachable except in test code. The one real use is `bumpRange`. But a range like `~2.0.0-internal.1.0.0` is detected as 'semver' and not internal.
To be explicit about my how does semver treat `~2.0.0-internal.1.0.0` in prior, that would become `>=2.0.0-internal.1.0.0 <2.1.0`.

Reply:
Are there cases now? Why aren't they getting fixed?
The comment I added is not really correct. But I don't think such cases should be treated as internal because the semver meaning of them is treat ~ and ^ as >=(given spec) to ^(given spec less prerelease), at least as far as I can ascertain.
I would say they should just be rejected - throw as invalid.